src/coreclr/src/jit/morph.cpp

```diff
@@ -13598,8 +13598,17 @@ GenTree* Compiler::fgMorphSmpOp(GenTree* tree, MorphAddrContext* mac)
 
             break;
 
-        case GT_NOT:
         case GT_NEG:
+            // Remove double negation
+            if ((oper == GT_NEG) && op1->OperIs(GT_NEG))
+            {
+                GenTree* child = op1->AsOp()->gtGetOp1();
+                return child;
+            }
+
+            __fallthrough;
+
+        case GT_NOT:
```

Comment:
Were you going to extend this to also cover `NOT(NOT(...)` ?

Reply:
Sure thing. Will do.